src/codeActionProvider.ts

```diff
@@ -0,0 +1,154 @@
+                return;
+            }
+            const annotationItems = allLombokAnnotations.map(name => {
+                return {
+                    label: `@${name}`,
+                    description: annotationsDescriptions[allLombokAnnotations.indexOf(name)],
+                    buttons: [{
+                        iconPath: new ThemeIcon("link-external"),
+                        tooltip: "Reference"
+                    }]
+                };
+            });
+            const initSelectItems = annotationItems.filter((item) => {
+                return annotationResponse.annotations.indexOf(item.label.split('@')[1]) >= 0;
+            });
+
+            const disposables: Disposable[] = [];
+            const selectItems = await new Promise<readonly QuickPickItem[]>(async (resolve, reject) => {
+                const pickBox = window.createQuickPick();
+                pickBox.items = annotationItems;
+                pickBox.canSelectMany = true;
+                pickBox.ignoreFocusOut = true;
+                pickBox.selectedItems = initSelectItems;
+                pickBox.placeholder = 'Select the Lombok Annotation';
+
+                disposables.push(
+                    pickBox.onDidTriggerItemButton(e => {
+                        const annotation = e.item.label.split('@')[1];
+                        env.openExternal(Uri.parse(annotationLinks[allLombokAnnotations.indexOf(annotation)]));
+                    }),
+                    pickBox.onDidAccept(() => {
+                        resolve(pickBox.selectedItems);
+                    }),
+                    pickBox.onDidHide(() => {
+                        resolve(pickBox.selectedItems);
+                    })
+                );
+                disposables.push(pickBox);
+                pickBox.show();
+            });
+            disposables.forEach(d => d.dispose());
+
+            selectAnnotations = selectItems.map(item => {
+                return item.label.split('@')[1];
+            });
+
+            delombokAnnotations = annotationResponse.annotations.filter((item) => {
+                return selectAnnotations.indexOf(item) < 0;
+            });
+            lombokAnnotations = selectAnnotations.filter((item) => {
+                return annotationResponse.annotations.indexOf(item) < 0;
+            });
+        }
+        if (delombokAnnotations.length === 0 && lombokAnnotations.length === 0) {
+            return;
+        }
+        const delombokParams: LombokRequestParams = {
+            context: params,
+            lombokAnnotations,
+            delombokAnnotations
+        };
+        const workspaceEdit = await executeJavaLanguageServerCommand(Commands.JAVA_CODEACTION_LOMBOK, JSON.stringify(delombokParams)) as WorkspaceEdit;
+        await applyWorkspaceEdit(workspaceEdit);
+        await revealWorkspaceEdit(workspaceEdit);
+    }));
+}
+
+
+export class LombokCodeActionProvider implements CodeActionProvider {
```

Comment:
```typescript
export interface CodeActionProvider<T extends CodeAction = CodeAction> {
        /**
         * Provide commands for the given document and range.
         *
         * @param document The document in which the command was invoked.
         * @param range The selector or range for which the command was invoked. This will always be a selection if
         * there is a currently active editor.
         * @param context Context carrying additional information.
         * @param token A cancellation token.
         *
         * @return An array of code actions, such as quick fixes or refactorings. The lack of a result can be signaled
         * by returning `undefined`, `null`, or an empty array.
         *
         * We also support returning `Command` for legacy reasons, however all new extensions should return
         * `CodeAction` object instead.
         */
        provideCodeActions(document: TextDocument, range: Range | Selection, context: CodeActionContext, token: CancellationToken): ProviderResult<(Command | T)[]>;

        /**
         * Given a code action fill in its {@linkcode CodeAction.edit edit}-property. Changes to
         * all other properties, like title, are ignored. A code action that has an edit
         * will not be resolved.
         *
         * *Note* that a code action provider that returns commands, not code actions, cannot successfully
         * implement this function. Returning commands is deprecated and instead code actions should be
         * returned.
         *
         * @param codeAction A code action.
         * @param token A cancellation token.
         * @return The resolved code action or a thenable that resolves to such. It is OK to return the given
         * `item`. When no result is returned, the given `item` will be used.
         */
        resolveCodeAction?(codeAction: T, token: CancellationToken): ProviderResult<T>;
    }
```

It's better to support lazy resolveCodeAction, this can save the CPU time a lot.